developer/src/tike/main/Keyman.System.KeymanCoreDebug.pas

```diff
@@ -106,6 +106,11 @@ km_kbp_state_debug_item = record
 
 km_kbp_debug_type = type uint32_t;
 
+// These types are used only for debugging convenience
+type
+  km_kbp_state_debug_item_array = array[0..100] of km_kbp_state_debug_item;
+  pkm_kbp_state_debug_item_array = ^km_kbp_state_debug_item_array;
+
```

Comment:
Where is this array and the one in `KeymanCore.pas` used?

Reply:
These are used only in the debugger. If you cast the relevant variable to `pkm_kbp_state_debug_item_array` type in the watch window, you can see the whole array rather than just the current item. It's annoying that the Delphi debugger doesn't do pointers to arrays well otherwise, in any way that I've been able to find!